lib/pages/add_page/widgets/recurrence_list_tile.dart

```diff
@@ -0,0 +1,140 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/cupertino.dart';
+import "package:flutter_riverpod/flutter_riverpod.dart";
+import 'dart:io' show Platform;
+
+import "../../../constants/style.dart";
+import '../../../providers/transactions_provider.dart';
+import '../../../model/transaction.dart';
+import 'recurrence_selector.dart';
+
+class RecurrenceListTile extends ConsumerWidget {
+  const RecurrenceListTile({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final isRecurring = ref.watch(selectedRecurringPayProvider);
+
+    return Column(
+      children: [
+        const Divider(height: 1, color: grey1),
+        ListTile(
+          contentPadding: const EdgeInsets.all(16),
+          leading: Container(
+            decoration: BoxDecoration(
+              shape: BoxShape.circle,
+              color: Theme.of(context).colorScheme.secondary,
+            ),
+            child: Padding(
+              padding: const EdgeInsets.all(10.0),
+              child: Icon(
+                Icons.autorenew,
+                size: 24.0,
+                color: Theme.of(context).colorScheme.background,
+              ),
+            ),
+          ),
+          title: Text(
+            "Recurring payment",
+            style: Theme.of(context)
+                .textTheme
+                .titleLarge!
+                .copyWith(color: Theme.of(context).colorScheme.primary),
+          ),
+          trailing: (Platform.isIOS)
```

Comment:
In dart exists some great adaptive widgets, switch is one of that, we should move to Switch.adaptive(), it will handle automatically the platform, using less code.

Reply:
Thanks, I'll change it to that